tests/basic-test.sh

```diff
@@ -809,8 +809,11 @@ cd ${test_tmpdir}
 rm -rf test2-checkout
 mkdir -p test2-checkout
 cd test2-checkout
-touch should-not-be-fsynced
-$OSTREE commit ${COMMIT_ARGS} -b test2 -s "Unfsynced commit" --fsync=false
+echo 'should not be fsynced' > should-not-be-fsynced
+if ! skip_one_without_strace_fault_injection; then
+    ${CMD_PREFIX} strace -o /dev/null -f -e inject=syncfs,fsync,sync:error=EPERM ostree --repo=${test_tmpdir}/repo commit ${COMMIT_ARGS} -b test2 -s "Unfsynced commit" --fsync=false
+    echo "ok fsync disabled"
+fi
```

Comment:
Maybe another test here for the opposite? I.e. checking that without `--fsync=false`, we fail here with `EPERM`.